core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseTelemetry.java

```diff
@@ -184,15 +207,15 @@ public void reset() {
     protected void setSampleRate(Envelope envelope) {
     }
 
-    protected String getEnvelopName() {
+    public String getEnvelopName() {
```

Comment:
Why do we need all of these as public....seems like we are exposing a lot more public APIs now. Can you explain the use case (as I might have overlooked on the glance) and can we try to reduce the exposure wherever possible.

Reply:
See this: https://github.com/Microsoft/ApplicationInsights-Java/blob/61ed8ac512838720e2066c3870661550c6ee6a09/core/src/main/java/com/microsoft/applicationinsights/channel/concrete/localforwarder/LocalForwarderModelTransformer.java#L133

Actually, I'm still on the fence about exposing these. I think there's another option. I'm going to experiment.